src/services/user-service.js

```diff
@@ -1,39 +1,51 @@
+const LRUCache = require('lru-native');
 const { log } = require('steno');
 
 const User = require('../models/UserModel');
-const cache = require('./cache-service.js');
 
-module.exports = {
-    get(uid) {
-        const cachedUser = cache.get(uid);
+const cache = new LRUCache({
+    maxElements: 1000,
+    maxAge: 120 * 60 * 1000, // 2 hours
+    size: 100
+});
+
+function getByUid(uid) {
+    const cachedUser = cache.get(uid);
+
+    if (cachedUser) {
+        return Promise.resolve(cachedUser);
+    }
 
-        if (cachedUser) {
-            return Promise.resolve(cachedUser);
+    // TODO: add step-by-step data retrieval:
+    //   1. all possible info from cache
+    //   2. rest from DB in one query
+    
+    return User.findOne({ uid }).exec().then((user) => {
+        if (user) {
+            cache.set(user.uid, user);
         }
 
-        return User.findOne({ uid }).exec().then((user) => {
-            if (user) {
-                cache.set(user.uid, user);
-            }
-        });
+        return user;
+    }); // TODO: add catch?
```

Comment:
I think it's okay to return `null` if we also handle all possible exceptions elsewhere.